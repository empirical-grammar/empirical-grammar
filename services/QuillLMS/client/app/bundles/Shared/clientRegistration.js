import 'lazysizes';
import 'lazysizes/plugins/parent-fit/ls.parent-fit';
import ReactOnRails from 'react-on-rails';
import './styles/styles.scss'

import {
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
} from './index'

ReactOnRails.register({
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
});
