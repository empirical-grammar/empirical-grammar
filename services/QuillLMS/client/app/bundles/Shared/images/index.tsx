export const bigCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-check-big.svg`,
  alt: "White checkmark in green circle"
}

export const searchMapIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-map-search.svg`,
  alt: "Map with a magnifying glass icon"
}

export const clipboardCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-clipboard-check.svg`,
  alt: "Clipboard with a checkmark icon"
}

export const tableCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-table-check.svg`,
  alt: "Table with a checkmark icon"
}

export const accountViewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-view.svg`,
  alt: "Avatar with an eye next to it icon"
}

export const giftIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-gift.svg`,
  alt: "Gift icon"
}

export const groupAccountIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-group.svg`,
  alt: "Three avatars icon"
}

export const googleClassroomIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-google-classroom.svg`,
  alt: "Google Classroom icon"
}
