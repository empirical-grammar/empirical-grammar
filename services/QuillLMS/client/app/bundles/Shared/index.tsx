export {
  QuestionList,
  LinkListItem,
  QuestionListByConcept
} from './components/questionList/index'

export {
  Instructions,
  Prompt
} from './components/fillInBlank/index'

export {
  ConceptExplanation,
} from './components/feedback/conceptExplanation'

export {
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  SortableList,
  TeacherPreviewMenu,
  TextEditor,
  TextArea,
  Tooltip,
  TwoThumbSlider,
  UploadOptimalResponses
} from './components/shared/index'

export {
  Modal
} from './components/modal/modal'

export {
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields
} from './components/questions/index'

export {
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments
} from './components/renderForQuestions/index'

export {
  QuestionRow
} from './components/scoreAnalysis/index'

export {
  MultipleChoice,
  Register,
  PlayTitleCard
} from './components/studentLessons/index'

export {
  TitleCard
} from './components/titleCards/index'

export {
  hashToCollection,
  isValidRegex,
  momentFormatConstants,
  copyToClipboard
} from './libs/index'
