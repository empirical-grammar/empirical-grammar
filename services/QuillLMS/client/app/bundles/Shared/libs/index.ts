export { hashToCollection } from './hashToCollection'
export { isValidRegex } from './isValidRegex'
export { momentFormatConstants } from './momentFormatConstants'
export { copyToClipboard } from './copyToClipboard'
