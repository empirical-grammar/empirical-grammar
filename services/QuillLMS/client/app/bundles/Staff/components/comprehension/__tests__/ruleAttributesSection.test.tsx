import * as React from 'react';
import { shallow } from 'enzyme';

import RuleAttributesSection from '../configureRules/ruleAttributesSection';
import RegexSection from '../configureRules/regexSection';
import { TextEditor } from '../../../../Shared/index'

const mockProps = {
  ruleType: 'Regex',
  plagiarismText: { text: '' },
  setPlagiarismText: jest.fn(),
  firstPlagiarismFeedback: { text: '' },
  setFirstPlagiarismFeedback: { text: '' },
  secondPlagiarismFeedback: { text: '' },
  setSecondPlagiarismFeedback: { text: ''},
  regexFeedback: { text: ''},
  setRegexFeedback: jest.fn(),
  errors: {},
  handleAddRegexInput: jest.fn(),
  handleDeleteRegexRule: jest.fn(),
  handleSetRegexRule: jest.fn(),
  regexRules: [{id: 1, regex_text: 'test'}],
};

describe('RuleAttributesSection component', () => {
  let container = shallow(<RuleAttributesSection {...mockProps} />);

  it('should render RuleAttributesSection', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render a TextEditor and RegexSection if ruleType is Regex', () => {
    // TextEditor: Feedback (1)
    // RegexSection (1)
    expect(container.find(TextEditor).length).toEqual(1);
    expect(container.find(RegexSection).length).toEqual(1);
  });
  it('should render a 2 TextEditor sections if ruleType is Plagiarism', () => {
    mockProps.ruleType = 'Plagiarism';
    container = shallow(<RuleAttributesSection {...mockProps} />);
    // TextEditor: Plagiarism Text, First Plagiarism Feedback, Second Plagiarism Feedback (3)
    expect(container.find(TextEditor).length).toEqual(3);
  });
});
