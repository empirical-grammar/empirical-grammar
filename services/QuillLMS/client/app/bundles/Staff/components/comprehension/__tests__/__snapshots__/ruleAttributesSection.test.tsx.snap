// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleAttributesSection component should render RuleAttributesSection 1`] = `
<Fragment>
  <p
    className="form-subsection-label"
  >
    Feedback
  </p>
  <TextEditor
    ContentState={[Function]}
    EditorState={[Function]}
    handleTextChange={[MockFunction]}
    key="regex-feedback"
    text=""
  />
  <p
    className="form-subsection-label"
    id="regex-rules-label"
  >
    Regex Rules
  </p>
  <RegexSection
    errors={Object {}}
    handleAddRegexInput={[MockFunction]}
    handleDeleteRegexRule={[MockFunction]}
    handleSetRegexRule={[MockFunction]}
    regexRules={
      Array [
        Object {
          "id": 1,
          "regex_text": "test",
        },
      ]
    }
  />
</Fragment>
`;
